client/main/dataset.js

```diff
@@ -118,11 +118,24 @@ const DataSetModel = Backbone.Model.extend({
 
         let columnPB = new coms.Messages.DataSetSchema.ColumnSchema();
         columnPB.id = id;
-        columnPB.name = values.name;
         columnPB.measureType = DataSetModel.parseMeasureType(values.measureType);
 
         let nameChanged = values.name !== column.name;
+        let newName = values.name;
+        let testName = newName;
+        let collision = 0;
         let oldName = column.name;
+        let columns = this.attributes.columns;
+        for (let i = 0; i < columns.length; i++) {
+            if (columns[i].name === testName) {
+                testName = newName + "(" + (collision + 2)  + ")";
+                collision += 1;
+                i = -1;
+            }
+        }
+        newName = testName;
```

Comment:
oic, the `i = -1` ... it's not a big deal, but it does go against people's expectations of a for-loop. if you're changing the value of i inside the loop, then a while loop is probably more appropriate (think of the for-loop as language).

can we get a space between the column name and the `(2)`?